docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+---
+title: Scrambled Transaction to combat against MEV
+---
+
+### Program Statement
+
+The existence of MEV isn't ideal and costly.
+
+MEV motivates transaction reordering, censoring and copying. Also, it causes
+cluster destabilization because of incentivised forking.
+
+The root problem is that there is no rule in the pursuit of MEV frontier, except
+economic rationality. While this anarchism is welcomed, stemmed from the
+blockchain philosophy, it's also desirable for such an activity to economically
+impractical in the first place.
+
+Any temporal market imbalance should be corrected strictly on the grounds of a
+fair game, basically the golden first-come-first-served rule (with optional
+bidding for priority).
+
+Note that this proposal doesn't fully mitigate one of MEV types, unlike all
+other fully-neutralized types: back-running. Back-running can be exercised
+while the golden rule is strictly and fairly enforced.
+
+### Summary of Proposed Solution
+
+This proposal tries to combat against MEV, at the cost of (slight) global
+induced latency of cluster as a whole.
+
+At highest level, a full mitigation of MEV can be attained _if leaders (and
+searchers) can't know the contents of other's transactions until finishing to
+packing and ordering them into a new block_. In addition to this, the
+transaction details must also be deterministically revealed after the ordering
+of transactions is committed.
+
+So, naive 2-step transaction submission (encrypted tx, then its key) won't work,
+let alone increased latency.
```

Comment:
Maybe you could clarify why naive 2-step transaction submission is doomed, is it because validators can't really wait for ALL the clients to send their 2nd part (the key) to decrypt transactions (and if so, what if we just drop those  transaction by clients that didn't manage to send their keys) ?

Reply:
>  is it because validators can't really wait for ALL the clients to send their 2nd part (the key) to decrypt transactions

yep.

> (and if so, what if we just drop those transaction by clients that didn't manage to send their keys) ?

if we allowed that, clients just won't send key if their original tx isn't favorable (arb turned out unprofitable).